xbmc/addons/kodi-dev-kit/include/kodi/c-api/addon_base.h

```diff
@@ -74,8 +75,12 @@
 #if __GNUC__ >= 4
 #define ATTR_DLL_IMPORT __attribute__((visibility("default")))
 #define ATTR_DLL_EXPORT __attribute__((visibility("default")))
+#ifndef SWIG
 #define ATTR_DLL_LOCAL __attribute__((visibility("hidden")))
 #else
+#define ATTR_DLL_LOCAL
+#endif
+#else
```

Comment:
Is this `#else` intended?

Reply:
Yes, looks a bit strange as this on github.
The `ATTR_DLL_LOCAL` only used on `__GNUC__ >= 4`.
Maybe with spaces on begin more clean:
```cpp
#if defined _WIN32 || defined _WIN64 || defined __CYGWIN__
  #define ATTR_DLL_IMPORT __declspec(dllimport)
  #define ATTR_DLL_EXPORT __declspec(dllexport)
  #define ATTR_DLL_LOCAL
  #ifdef _WIN64
    #define ATTR_APIENTRY __stdcall
  #else
    #define ATTR_APIENTRY __cdecl
  #endif
#else
  #if __GNUC__ >= 4
    #define ATTR_DLL_IMPORT __attribute__((visibility("default")))
    #define ATTR_DLL_EXPORT __attribute__((visibility("default")))
    #ifndef SWIG
      #define ATTR_DLL_LOCAL __attribute__((visibility("hidden")))
    #else
      #define ATTR_DLL_LOCAL
    #endif
  #else
    #define ATTR_DLL_IMPORT
    #define ATTR_DLL_EXPORT
    #define ATTR_DLL_LOCAL
  #endif
  #define ATTR_APIENTRY
#endif
```